controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -998,24 +1032,23 @@ private boolean verifyRecordMatchesInput(List<Long> segmentsToSeal, List<SimpleE
             } else if (ex != null) {
                 throw new CompletionException(ex);
             }
-            return CompletedTxnRecord.parse(ok.getData()).getCompletionStatus();
+            return CompletedTxnRecord.fromBytes(ok.getData()).getCompletionStatus();
         });
     }
 
     @Override
     public CompletableFuture<SimpleEntry<TxnStatus, Integer>> sealTransaction(final UUID txId, final boolean commit,
                                                                               final Optional<Version> version) {
-        val legal = verifyLegalState();
```

Comment:
Where do we do this verification now?

Reply:
Including `verifyLegalState` method call means an additional call to the store to fetch the state. 

So i have removed it because it was more of a sanity check as we cannot ever get into a bad state here. 
Reason: all transactions are taken care of before a stream is sealed. So for a sealed stream there would be no active transactions. 
Similarly for a stream in Creating state, there cannot be any transaction created. 
All other intermediate states are legitimate for sealTransaction to be called. 

So verifyLegalState method was an unnecessary overhead.